syntax = "proto3";

package scanner_ai;

enum AIResult {
  AI_RESULT_SUCCESS_UNSPECIFIED = 0;
  AI_RESULT_ERROR_UNKNOWN = 1;
}

message AIInitReq {
}

message AIInitRes {
  AIResult result = 1;
}

message AIPredictSampleReq {
  uint32 id = 1;
  uint32 sampling_rate_khz = 2;
  uint32 center_freq_khz = 3;
  uint32 bw_khz = 4;
  uint32 num_chunks = 5;
  uint32 overlay_khz = 6;
  uint32 samples_len = 7;
}

message AIPredictSampleRes {
  uint32 id = 1;
  AIResult result = 2;
}

message AISampleDataReq {
  uint32 id = 1;
  repeated float samples = 2;
}

message AISampleDataRes {
  uint32 id = 1;
  repeated float lte_freqs = 2;
  repeated float umts_freqs = 3;
  repeated float gsm_freqs = 4;
}

message AIModelReq {
  oneof message {
    AIInitReq init_req = 1;
    AIPredictSampleReq predict_sample_req = 2;
    AISampleDataReq sample_data_req = 3;
  }
}

message AIModelRes {
  oneof message {
    AIInitRes init_res = 1;
    AIPredictSampleRes predict_sample_res = 2;
    AISampleDataRes sample_data_res = 3;
  }
}
